### PR TITLE
Integration tests for s3_logging

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

SSIA: add integration tests from s3_logging
The tests are pretty basic, it doesn't provide any form of output beyond returning changed

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/amazon/s3_logging.py
test/sanity/ignore.txt
hacking/aws_config/testing_policies/storage-policy.json

##### ADDITIONAL INFORMATION
Depends on https://github.com/mattclay/aws-terminator/pull/65

### CODE REVIEW

**Review by jillr — DISMISSED** [review]

**Comment by jillr** [comment]
The tests pass when I run them locally against CI but are failing in shippable, will have to dig into this more.

File: test/integration/targets/s3_logging/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      state: present
+      name: '{{ test_bucket }}'
+      target_bucket: '{{ log_bucket_2 }}'
+      target_prefix: '/{{ resource_prefix }}/'
+    register: result
+
+  - assert:
+      that:
+      - result is changed
+
```
Comment (jillr) [inline test/integration/targets/s3_logging/tasks/main.yml:128]: Here, 15s is working for me

File: test/integration/targets/s3_logging/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    s3_bucket:
+      state: present
+      name: '{{ log_bucket_2 }}'
+    register: output
+
+  - assert:
+      that:
+      - output is changed
+      - output.name == log_bucket_2
+
```
Comment (jillr) [inline test/integration/targets/s3_logging/tasks/main.yml:66]: Needs a pause here, seems to need 30s

File: test/integration/targets/s3_logging/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  - name: Disable logging
+    s3_logging:
+      state: absent
+      name: '{{ test_bucket }}'
+    register: result
+
+  - assert:
+      that:
+      - result is changed
+
```
Comment (jillr) [inline test/integration/targets/s3_logging/tasks/main.yml:176]: And here for 15s

**Review by jillr — COMMENTED** [review]
Looks like the CI failures are related to eventual consistency. Adding pauses for the specified times at the noted locations in the test seemed to be working.

**Comment by tremble** [comment]
> The tests pass when I run them locally against CI but are failing in shippable, will have to dig into this more.

Out of curiosity, which region do you 'locally' test on.  I was running in us-west-2 and had no problems...

File: test/integration/targets/s3_logging/tasks/main.yml (hunk 1 above)
Comment (tremble) [reply test/integration/targets/s3_logging/tasks/main.yml:128]: Thanks for digging into that @jillr

**Comment by jillr** [comment]
> > The tests pass when I run them locally against CI but are failing in shippable, will have to dig into this more.
> 
> Out of curiosity, which region do you 'locally' test on. I was running in us-west-2 and had no problems...

@tremble, also us-west-2 but it was showing up intermittently - if CI hadn't failed I would have probably missed it to be honest. I have a simple bash script I use for things like this that loops 50x over a test and once I started looping this one it was coming up maybe 1 in 5 times, roughly.

**Comment by jillr** [comment]
Hm, still getting occasional failures when looping at:
```
TASK [s3_logging : Enable logging] *********************************************
fatal: [testhost]: FAILED! => {"changed": false, "msg": "You must give the log-delivery group WRITE and READ_ACP permissions to the target bucket"}
```
These tests are likely to be unstable as-is, will need to dive into this again as the pauses might not be what we need after all.

**Review by jillr — APPROVED** [review]
shipit

**Pull request merged**